### ISSUE TITLE
Error in assembly process

### ISSUE THREAD

**Issue by yroskov** [CatalogueOfLife/clearinghouse-ui#713]
Description of steps:
Doing assembly of World Plants, I have (1) deleted all sectors in order Cucurbitales, (2) deleted subtree Cucurbitales, (3)  drag&dropped order Cucurbitales, set up it as a sector, synced (**sync process looked as completed: no spinning weals, status label: Idle**). 

![image](https://user-images.githubusercontent.com/28574252/92282012-6d1e7480-eec2-11ea-99e2-d90a88d20426.png)

(4) Then, I need to add family Datiscaceae from another place of World Plants in a newly created sector Cucurbitales; drag&dropped family Datiscaceae as a child. But got error messages:

![image](https://user-images.githubusercontent.com/28574252/92267725-bf519c80-eea6-11ea-8fc5-12d68ce33ef8.png)

![image](https://user-images.githubusercontent.com/28574252/92267865-ff188400-eea6-11ea-9604-4e18aade3450.png)

Later (??mins) the problem autoresolved. Repeated steps came to success.

**Comment by gdower** [comment]
I'm pretty sure this is a frontend bug, because after refreshing the full browser window it resolves the issue for Yuri. It seems like maybe the IDs are out of sync.

**Comment by mdoering** [comment]
Yes it is. When a new sector is created the backend also creates a few taxa immediately so the UI can visualize the new sector. Thats what you saw on creation. When you then sync the new sector all records of that sector get removed and new ones copied in with new temporary UUIDs. I assume the UI still kept the first created taxa around that you used to drop the subsequent sector on. But which did not exist at that time anymore. 

Does the UI not remove all sector children when a sync is triggered?

**Comment by gdower** [comment]
@mdoering, I wonder if it is caused by longer running sync jobs that block sector deletion from completing? So if @yroskov was syncing a large sector, then deleted a different sector, would the large sector have to finish syncing before the other sector gets deleted? If so, then when he tried to assemble a new sector the other sector might have still been pending deletion.

**Comment by mdoering** [comment]
Yes, deletions are queued together with syncs and currently we allow to run just one in parallel for a given project as they might interfere with each other.

We need a better job management both in the [API](https://github.com/CatalogueOfLife/backend/issues/721) and [UI](https://github.com/CatalogueOfLife/clearinghouse-ui/issues/337). I have started that for the background admin jobs, but we need to merge imports, syncs and deletions to also use this single async job queue system and provide a more generic and better UI for all these background jobs. Then we can show them anywhere, filter to relevant projects or even users only and be better informed whats still queued or currently running.

@yroskov I would recommend not to modify the same part of the tree when there are currently running syncs or deletions.

**Comment by yroskov** [comment]
>I would recommend not to modify the same part of the tree when there are currently running syncs or deletions.
It is serious bottleneck in editorial process. If I need to delete 10 sectors and 10 families in the same order, slow deletions and slow syncs make all steps very slow and confusing (operations blocked without explanations; reaction of the system looks inconsistent). 

How to optimize a code and make deletions & syncs fast enough?

**Comment by mdoering** [comment]
I cant think of anything to accelerate the process. Syncs do happen from top to bottom, so you can already attach new sectors to top parts of the tree while the bulk of lower ranked taxa like species still syncs.
